File: Lab7 - Spring - Java/src/main/java/pl/edu/agh/to/school/course/Course.java
package pl.edu.agh.to.school.course;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import pl.edu.agh.to.school.grade.Grade;
import pl.edu.agh.to.school.student.Student;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Course {
    @Id
    @GeneratedValue
    private int id;
    private String name;
    @OneToMany
    private List<Student> students;

    @OneToMany(mappedBy = "course", orphanRemoval = true)
    private List<Grade> gradesList = new ArrayList<>();

    public Course(String name) {
        this.name = name;
        this.students = new ArrayList<>();
    }

    public Course() {
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void assignStudent(Student student) {
        students.add(student);
    }

    public List<Student> getStudents() {
        return students;
    }

    public List<Grade> getGradesList() {
        return gradesList;
    }

    public void addGrade(Grade grade) {
        gradesList.add(grade);
    }
}

File: Lab7 - Spring - Kotlin/src/main/kotlin/pl/edu/agh/to/school/course/Course.kt
package pl.edu.agh.to.school.course

import pl.edu.agh.to.school.student.Student

class Course(
    val name: String,
    val students: ArrayList<Student>
) {
    fun getStudents(): List<Student> {
        return students
    }

    fun assignStudents(student: Student) {
        students.add(student)
    }

    fun removeStudent(student: Student) {
        students.remove(student)
    }
}